src/mail-app/mail/export/MailExportFacade.ts

```diff
@@ -0,0 +1,122 @@
+import { MailExportTokenService } from "../../../common/api/entities/tutanota/Services"
+import { createMailExportTokenServicePostIn } from "../../../common/api/entities/tutanota/TypeRefs"
+import { assertWorkerOrNode } from "../../../common/api/common/Env"
+import { IServiceExecutor } from "../../../common/api/common/ServiceRequest"
+import { EntityClient } from "../../../common/api/common/EntityClient"
+import { CacheMode, EntityRestClientLoadOptions } from "../../../common/api/worker/rest/EntityRestClient"
+import type { ListElementEntity, SomeEntity } from "../../../common/api/common/EntityTypes"
+import { TypeRef } from "@tutao/tutanota-utils"
+import { AccessExpiredError } from "../../../common/api/common/error/RestError"
+import { resolveTypeReference } from "../../../common/api/common/EntityFunctions"
+import { ValueType } from "../../../common/api/common/EntityConstants"
+import { CUSTOM_MIN_ID, elementIdPart, GENERATED_MIN_ID, getLetId, RANGE_ITEM_LIMIT } from "../../../common/api/common/utils/EntityUtils"
+
+assertWorkerOrNode()
+
+const TAG = "MailExportFacade"
+
+export type ExportToken = string & { _exportToken: undefined }
+
+export class MailExportFacade {
+	// Mail group ID -> export token
+	private readonly tokenCache: Map<Id, ExportToken> = new Map()
+
+	constructor(private readonly serviceExecutor: IServiceExecutor, private readonly entityClient: EntityClient) {}
+
+	/**
+	 * Load a single element for export, (re-)generating a mail export token if needed
+	 * @param mailGroup
+	 * @param typeRef
+	 * @param id
+	 */
+	async loadForMailGroup<T extends SomeEntity>(mailGroup: Id, typeRef: TypeRef<T>, id: PropertyType<T, "_id">): Promise<T> {
+		return await this.handleRequest(mailGroup, async (options) => {
+			return await this.entityClient.load(typeRef, id, options)
+		})
+	}
+
+	/**
+	 * Load a list of elements for export, (re-)generating a mail export token if needed
+	 */
+	async loadAllForMailGroup<T extends ListElementEntity>(mailGroup: Id, typeRef: TypeRef<T>, listId: Id, start?: Id): Promise<T[]> {
+		const typeModel = await resolveTypeReference(typeRef)
+
+		if (!start) {
+			start = typeModel.values["_id"].type === ValueType.GeneratedId ? GENERATED_MIN_ID : CUSTOM_MIN_ID
+		}
+
+		const allElements: T[] = []
+		let firstId = start
+		while (true) {
+			// Don't want to use entityClient.loadAll in case the token expires in the middle of us getting entities and
+			// then we'd have to do it all over again...
+			const elements = await this.handleRequest(mailGroup, async (options) => {
```

Comment:
we need to load and store them chunk by chunk so `loadAll` doesn't make sense anyway